### ISSUE TITLE
[Lint] `N804` Stop enforcing `cls` for metaclasses' `@classmethod`s first argument

### ISSUE THREAD

**Issue by eliegoudout** [astral-sh/ruff#15143]
Hello,

There are many discussions (_e.g._ #13698) about metaclasses and `N804`, some of which may require further inquieries. But I'm fairly certain we can already agree on the fact that **enforcing `cls` for the first argument name of a metaclass' `@classmethod` if wrong**.

```python
class A(type):
    def a(self): ...  # Should FAIL: OK

    def b(cls): ...  # Should PASS: OK

    @classmethod
    def c(cls): ...  # Should FAIL: NOT OK

    @classmethod
    def d(mcs): ...  # Should PASS: NOT OK
```

Output:
```
tmp.py:2:11: N804 First argument of a class method should be named `cls`
  |
1 | class A(type):
2 |     def a(self): ...
  |           ^^^^ N804
3 | 
4 |     def b(cls): ...
  |
  = help: Rename `self` to `cls`

tmp.py:10:11: N804 First argument of a class method should be named `cls`
   |
 9 |     @classmethod
10 |     def d(mcs): ...
   |           ^^^ N804
   |
   = help: Rename `mcs` to `cls`
```

**Comment by MichaReiser** [comment]
Thanks for opening this issue. I'd prefer if we keep the discussion in as few places as possible and there's already https://github.com/astral-sh/ruff/issues/6017 (which also explains some of the reasons why it is the way it is today, not saying we shouldn't change).
